app/models/ems_refresh/save_inventory.rb

```diff
@@ -101,7 +102,11 @@ def save_vms_inventory(ems, hashes, target = nil)
         found.raw_power_state = key_backup[:raw_power_state]
 
         link_habtm(found, key_backup[:storages], :storages, Storage)
-        link_habtm(found, key_backup[:security_groups], :security_groups, SecurityGroup)
+        # TODO(lsmola) NetworkManager, once all providers are converted, security groups relation of Vm will go away,
+        # being moved to NetworkPort, we will ignore here through associations
+        if found.try(:security_groups).try(:proxy_association).try(:reflection).try(:options).try(:[], :through) != :network_ports
```

Comment:
@Ladas that one starts from the collection; you're calling it just to get the reflection. It wasn't more bad, just more indirect.

The particular badness, beyond poking around associations at all (the linked example is different, because it's designed to be generic), was the long chain of `try` calls, when most of them weren't "genuine" -- only a couple of them could actually fail on their own. My alternative above gets that down to a single possibly-nil value.